individual-reflection.md

```diff
@@ -9,48 +9,61 @@ Answer the following comprehension questions **within this file.** Write your an
 1. What are the query params (the additional data sent with the request, besides the verb and the path)?
-    - Answer: 
+    - Answer: the query params are additional data that are sent with the request along with the path. For example, Token is an addtional required parameter that we need to pass in along with the url path. 
 1. What is the syntax used to make this request? (Copy and paste a code snippet here)
     - Answer:
       ```ruby
-      # Copy and paste your answer below this comment
+      def self.get(url)
+        response = HTTParty.get(url, query: {token: ENV['SLACK_TOKEN']})
+
+        if response.code != 200 || response["ok"] == false
+      raise SlackAPIError, "We encountered a problem: #{response["error"]}"
+        end
 
-      # Copy and paste your answer above this comment
+        return response
+        end
       ```
 1. What does the program do if the response comes back with a status code of 200?
-    - Answer: 
+    - Answer: OK/success
 1. What does the program do if the response does not come back with a status code of 200?
-    - Answer: 
+    - Answer: ["ok"] == false
 
 ### `POST` Request Review
 
 If your project does not make a POST request, read through Wave 3 on the original Slack CLI, and research and answer questions 1, 2, 3, 4, 6, and 7.
 
 1. Describe a POST request that your project makes, and the high-level description of what it does
-    - Answer:
+    - Answer: POSt request is the client tells the server that here is some data, please save it. 
 1. What is the verb of this request?
-    - Answer:
+    - Answer: POST
 1. What is the path (or the URL, or endpoint) of this request?
-    - Answer:
+    - Answer:url = "https://slack.com/api/chat.postMessage"
 1. What are the query params (the additional data sent with the request, besides the verb and the path)?
-    - Answer: 
+    - Answer: query: {token: ENV['BOT_TOKEN'], channel: self.slack_id, text: msg_text})
 1. What is the syntax used to make this request? (Copy and paste a code snippet here)
     - Answer:
       ```ruby
-      # Copy and paste your answer below this comment
-
-      # Copy and paste your answer above this comment
+        def send_message(msg_text)
+            url = "https://slack.com/api/chat.postMessage"
+            response = HTTParty.post(url, query: {token: ENV['BOT_TOKEN'], channel: self.slack_id, text: msg_text})
+
+            if response.code != 200 || response["ok"] == false
+                raise SlackAPIError, "We encountered a problem: #{response["error"]}"
+            else
+            return true
+        end
+        
       ```
 1. What does the program do if the response comes back with a status code of 200?
-    - Answer: 
+    - Answer: OK or Sucess
 1. What does the program do if the response does not come back with a status code of 200?
-    - Answer: 
+    - Answer: ["ok"] == false
```

Comment:
Same as above: This line describes how to check if a status code is not 200 (also, the syntax doesn't quite make sense. In Ruby, `["ok"]` describes an array that holds the string `"ok"` as the firs telement). What does your Ruby code do  in this case?